### PR TITLE
add docs for parallel and root hooks

### PR DESCRIPTION
**Pull request by boneskull**
> (this PR needs the changes from the `boneskull/issue/2839-11ty-changes` branch)

Build might not work until that one gets merged

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1233,6 +1262,288 @@ Prepend `--v8-` to any flag listed in the output of `node --v8-options` (excludi
...
+
+It's unlikely (but not impossible) to see a performance gain from a [job count](#-jobs-count-j-count) _greater than_ the number of available CPU cores. That said, _play around with the job count_--there's no one-size-fits all, and the unique characteristics of your tests will determine the optimal number of jobs; it may even be that fewer is faster!
+
+## Root Hook Plugins
+
+> _New in v8.0.0._
+
+In some cases, you may want to execute a [hook](#hooks) before (or after) every test in every file. Previous to v8.0.0, the way to accomplish this was to use `--file` combined with root-level hooks (see [example above](#root-hooks-are-not-global)). This still works in v8.0.0, but _not_ when running tests in parallel mode!
+
+A Root-Level Hook Plugin is a JavaScript file loaded via [`--require`](#-require-module-r-module) which "registers" one or more root-level hooks to be used across all test files.
```
Comment (craigtaub) [inline docs/index.md:1366]: Some users might go straight to `(#-require-module-r-module)` section, worth making a small update there to point here? Thinking the `"Note you cannot use --require to set a global beforeEach() hook"` bit could do with an update.

Comment (boneskull) [reply docs/index.md:1366]: yes, good catch.  I'll update it... want to see how #4244 actually lands first though

File: docs/index.md [hunk 2]
```diff
@@ -1233,6 +1262,288 @@ Prepend `--v8-` to any flag listed in the output of `node --v8-options` (excludi
...
+
+Some types of tests are _not_ so well-suited to run in parallel. For example, extremely timing-sensitive tests, or tests which make I/O requests to a limited pool of resources (such as opening ports, or automating browser windows, hitting a test DB, or remote server, etc.).
+
+Free-tier cloud CI services may not provide a suitable multi-core container or VM for their build agents. Regarding expected performance gains in CI: your mileage may vary. It may help to use a conditional in a `.mocharc.js` to check for `process.env.CI`, and adjust the job count as appropriate.
+
+It's unlikely (but not impossible) to see a performance gain from a [job count](#-jobs-count-j-count) _greater than_ the number of available CPU cores. That said, _play around with the job count_--there's no one-size-fits all, and the unique characteristics of your tests will determine the optimal number of jobs; it may even be that fewer is faster!
+
+## Root Hook Plugins
+
+> _New in v8.0.0._
```
Comment (nicojs) [inline docs/index.md:1362]:
```suggestion
> _New in v7.2.0._
```

File: docs/index.md [hunk 3]
```diff
@@ -1233,6 +1262,288 @@ Prepend `--v8-` to any flag listed in the output of `node --v8-options` (excludi
...
+```
+
+When run (in the default "serial" mode) via `mocha --file "./test/setup.js" "./test/**/*.spec.js"`, `setup.js` will be executed _first_, and install the two hooks shown above for every test found in `./test/**/*.spec.js`.
+
+**When Mocha runs in parallel mode, test files do not share the same process.** Consequently, a root-level hook defined in test file _A_ won't be present in test file _B_.
+
+There are a (minimum of) two workarounds for this:
+
+1. `require('./setup.js')` or `import './setup.js'` at the top of every test file. Best avoided for those averse to boilerplate.
+1. _Recommended_: Define root-level hooks in a required file, using the new (also as of v8.0.0) [Root Hook Plugin](#root-hook-plugins) system.
```
Comment (nicojs) [inline docs/index.md:1333]:
```suggestion
1. _Recommended_: Define root-level hooks in a required file, using the new (as of v7.2.0) [Root Hook Plugin](#root-hook-plugins) system.
```

File: docs/index.md (hunk 2 above)
Comment (boneskull) [reply docs/index.md:1268]: I think we need to discuss whether we want to do this in v7.2.0 or v8.0.0.  I would prefer the latter, for marketing reasons. 😜

**Comment by boneskull** [comment]
kept these references as v8.0.0 above, and changed a reference to v7.3.0 to v8.0.0

**Pull request merged**